tests/zfs-tests/tests/functional/cli_root/zinject/zinject_probe.ksh

```diff
@@ -0,0 +1,65 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or https://opensource.org/licenses/CDDL-1.0.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2025, Klara, Inc.
+#
+
+. $STF_SUITE/include/libtest.shlib
+
+verify_runnable "global"
+
+log_assert "Check zinject can correctly inject a probe failure."
+
+DISK1=${DISKS%% *}
+
+function cleanup
+{
+	zinject -c all
+	default_cleanup_noexit
+}
+
+log_onexit cleanup
+
+zpool create $TESTPOOL $DISK1
+
+log_must zinject -d $DISK1 -e io -T probe $TESTPOOL
+log_must zinject -d $DISK1 -e io -T write $TESTPOOL
+
+zinject
+
+log_must dd if=/dev/urandom of=/$TESTPOOL/file bs=1M count=1
+
+log_note "waiting for pool to suspend"
+typeset -i tries=10
+until [[ $(cat /proc/spl/kstat/zfs/$TESTPOOL/state) == "SUSPENDED" ]] ; do
```

Comment:
This is Linux'ism.  On FreeBSD it is in sysctls.

Reply:
This might work for cross-platform: `[[ $(get_pool_prop health $pool) == "SUSPENDED" ]]`

I dunno if you're looking at the kstat for a lockless lookup though...

Reply:
Yeah, lockless - ioctls are fickle when the pool is suspended.

We have a `kstat` method in libtest that is almost the right thing, and we barely use it. Leave it with me; I'm gonna spend my morning generalising it for use on a pool kstat too, and then making sure we use it everywhere - there's a bunch of places we don't.

Reply:
Posted #16950 with a new `kstat` helper, and rebased this to use it.

Reply:
Thanks, but I don't see `zinject_probe.ksh` to be updated.

Reply:
Ahh, I didn't actually commit it, just pushed. Sigh. Try now :)